creator-node/src/middlewares.js

```diff
@@ -762,14 +762,21 @@ async function getReplicaSetSpIDs({
 
       try {
         // will throw error if blocknumber not found
-        replicaSet =
-          await libs.contracts.UserReplicaSetManagerClient.getUserReplicaSetAtBlockNumber(
-            userId,
-            blockNumber
-          )
-        errorMsg = null
-        blockNumberIndexed = true
-        break
+        const encodedUserId = libs.Utils.encodeHashId(userId)
+        const spResponse = await libs.discoveryProvider.getUserReplicaSet({
+          encodedUserId,
+          blockNumber
+        })
+
+        if (spResponse) {
```

Comment:
given the repeated code here (also the if block is slightly divergent, i assume unintentionally) moving this to a util would be good - can just put it in this file for now